internal/datastore/postgres/postgres.go

```diff
@@ -118,14 +126,27 @@ func NewPostgresDatastore(
 		}
 	}
 
-	return &pgDatastore{
+	datastore := &pgDatastore{
 		dburl:                     url,
 		dbpool:                    dbpool,
 		watchBufferLength:         config.watchBufferLength,
 		revisionFuzzingTimedelta:  config.revisionFuzzingTimedelta,
 		gcWindowInverted:          -1 * config.gcWindow,
+		gcInterval:                config.gcInterval,
+		gcMaxOperationTime:        config.gcMaxOperationTime,
 		splitAtEstimatedQuerySize: config.splitAtEstimatedQuerySize,
-	}, nil
+		disposed:                  make(chan struct{}),
```

Comment:
since this is the only datastore than needs to GC anything what about injecting the channel when constructing it?

i.e.:
```go
func main() {
  done := make(chan struct{}) 
  ds := NewPostgresDatastore("url", WithDone(done))
  defer close(done)
}
```

that way you don't have to have `Dispose()` method in the interface (this approach can also close the channel when the process is killed, which I don't see happening right now)

Reply:
return an error if `WithDone` isn't given?

or we could have datastores take a context, and any datastore that needs to can listen for context.Done?

Reply:
`WithDone` would only be required by the postgres ds constructor in that case, though. And context still may be a better fit?

Reply:
Injecting a channel is very weird. I think we should just make the API similar to how we're managing the lifecycles of http and gRPC servers in the cobra runfuncs: a simple Close() or Shutdown() method that blocks (I've seen them take contexts sometimes too for configuring deadlines).

Inside the postgres implementation we should probably use errgroup and context cancellation rather than a chan of empty struct.

Reply:
@jzelinskie That's what `Dispose` does (minus using a chan vs an errgroup). I preferred `Dispose` for the name as it might be also cleaning up data too (in the potential sqlite case if we ended up using a file)